rmgpy/data/kinetics/database.py

```diff
@@ -400,37 +404,83 @@ def generateReactionsFromLibrary(self, reactants, products, library):
-                reactionList.append(reaction)
-        if products:
-            reactionList = filterReactions(reactants, products, reactionList)
-        return reactionList
-
-    def generateReactionsFromFamilies(self, reactants, products, only_families=None):
-        """
-        Generate all reactions between the provided list of one or two
-        `reactants`, which should be :class:`Molecule` objects. This method
-        applies the reaction family.
-        If `only_families` is a list of strings, only families with those labels
-        are used.
-        """
-        # If there are two structures and they are the same, then make a copy
-        # of the second one so we can independently manipulate both of them 
-        # This is for the case where A + A --> products
-        if len(reactants) == 2 and reactants[0] == reactants[1]:
-            reactants[1] = reactants[1].copy(deep=True)
-        
-        reactionList = []
+                reaction_list.append(reaction)
+
+        return reaction_list
+
+    def generate_reactions_from_families(self, reactants, products=None, only_families=None, resonance=True, clear_labels=True):
+        """
+        Generate all reactions between the provided list or tuple of one or two
+        `reactants`, which can be either :class:`Molecule` objects or :class:`Species`
+        objects. This method can apply all kinetics families or a selected subset.
+
+        Args:
+            reactants:      Molecules or Species to react
+            products:       List of Molecules or Species of desired product structures (optional)
+            only_families:  List of family labels to generate reactions from (optional)
+                            Default is to generate reactions from all families
+            resonance:      Flag to generate resonance structures for reactants and products (optional)
+                            Default is True, resonance structures will be generated
+            clear_labels:   Flag to clear atom labels from reactants after reaction generation (optional)
+                            Default is True, labels will be cleared
+
+        Returns:
+            List of reactions containing Species objects with the specified reactants and products.
+        """
+        # Check if the reactants are the same
+        # If they refer to the same memory address, then make a deep copy so
+        # they can be manipulated independently
+        same_reactants = False
+        if len(reactants) == 2:
+            if reactants[0] is reactants[1]:
+                reactants[1] = reactants[1].copy(deep=True)
+                same_reactants = True
+            elif reactants[0].isIsomorphic(reactants[1]):
+                same_reactants = True
+
+        # Convert to Species objects if necessary
+        reactants = ensure_species(reactants)
+
+        # Label reactant atoms for proper degeneracy calculation
+        ensure_independent_atom_ids(reactants, resonance=resonance)
```

Comment:
if `resonance =True`, this method only generate resonance isomers when ids are not independent, possible non-deterministic behavior?

Reply:
You're right, I updated it to also generate resonance structures if the IDs are already independent.